fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java

```diff
@@ -155,6 +162,60 @@ public SeekableInputStream toSeekableInputStream() {
         public boolean isExistOnDisk() {
             return this.inputFile != null && this.inputFile.exists();
         }
+
+        public void pin() {
+            useCount += 1;
+        }
+        public void unpin() {
+            useCount -= 1;
+        }
+
+        public static DiskCacheEntry newDiskCacheEntry(SeekableInputStream stream, long fileLength, String key) {
+            long totalBytesToRead = fileLength;
+            HadoopOutputFile tmpOutputFile = (HadoopOutputFile) IOUtil.getTmpOutputFile(
+                    METADATA_CACHE_DISK_PATH, key);
+            PositionOutputStream outputStream = tmpOutputFile.createOrOverwrite();
```

Comment:
why overwrite if exist

Reply:
it's tmp file, it's existing means we encountered exception when write rename and delete this file, so the file may have some errors.